test/unit/puppetlabs/services/jruby_pool_manager/jruby_ref_pool_test.clj

```diff
@@ -131,3 +133,36 @@
             pill (ShutdownPoisonPill. pool)]
         ; Returning a pill should be a noop
         (jruby-core/return-to-pool pool-context pill :test [])))))
+
+(deftest shutdown-prevents-flush
+  (testing "Triggering a flush after shutdown has been requested does not break shutdown"
+    (let [config (jruby-test-config 0 2)
+          ;; Set up the pool manually
```

Comment:
Could be good to say *why* you're doing it manually.

Reply:
Good point. It was a weird thing actually.